src/expressions.md

```diff
@@ -327,96 +429,120 @@ let y = 0...10;
 
 * `-`
   : Negation. Signed integer types and floating-point types support negation. It
     is an error to apply negation to unsigned types; for example, the compiler
-    rejects `-1u32`.
+    rejects `-1u32`. `-` can be overloaded for a type by implementing the
+    `std::ops::Neg` trait.
 * `*`
   : Dereference. When applied to a [pointer](types.html#pointer-types) it
     denotes the pointed-to location. For pointers to mutable locations, the
     resulting [lvalue](expressions.html#lvalues-rvalues-and-temporaries) can be
-    assigned to.  On non-pointer types, it calls the `deref` method of the
-    `std::ops::Deref` trait, or the `deref_mut` method of the
-    `std::ops::DerefMut` trait (if implemented by the type and required for an
-    outer expression that will or could mutate the dereference), and produces
-    the result of dereferencing the `&` or `&mut` borrowed pointer returned
-    from the overload method.
+    assigned to. Dereferencing a raw pointer requires `unsafe`. On non-pointer
+    types `*x` is equivalent to `*std::ops::Deref::deref(&x)` or
+    `*std::ops::Deref::deref_mut(&mut x)` depending on whether the rquired
+    lvalue has to be mutable.
 * `!`
   : Logical negation. On the boolean type, this flips between `true` and
-    `false`. On integer types, this inverts the individual bits in the
-    two's complement representation of the value.
+    `false`. On integer types, this inverts the individual bits of the binary
+    representation of the value, using two's complement for signed integers.
+    `!` can be overloaded for a type by implementing the `std::ops::Not` trait.
 * `&` and `&mut`
-  : Borrowing. When applied to an lvalue, these operators produce a
-    reference (pointer) to the lvalue. The lvalue is also placed into
-    a borrowed state for the duration of the reference. For a shared
-    borrow (`&`), this implies that the lvalue may not be mutated, but
-    it may be read or shared again. For a mutable borrow (`&mut`), the
-    lvalue may not be accessed in any way until the borrow expires.
-    If the `&` or `&mut` operators are applied to an rvalue, a
-    temporary value is created; the lifetime of this temporary value
-    is defined by [syntactic rules](#temporary-lifetimes).
+  : Borrowing. When applied to an lvalue, these operators produce a reference
+    (pointer) to the lvalue. The lvalue is also placed into a borrowed state
+    for the duration of the reference. For a shared borrow (`&`), this implies
+    that the lvalue may not be mutated, but it may be read or shared again. For
+    a mutable borrow (`&mut`), the lvalue may not be accessed in any way until
+    the borrow expires. If the `&` or `&mut` operators are applied to an
+    rvalue, a temporary value is created; the lifetime of this temporary value
+    is defined by [syntactic rules](#temporary-lifetimes). `&mut` may only be
+    applied to `lvalues` that can be mutated. These operators cannot be
```

Comment:
I want to write 'mutable lvalue' so much, but is that the proper way to say what I mean?